contracts/prover/BaseProver.sol

```diff
@@ -11,62 +12,113 @@ import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
-     * @dev Zero address indicates intent hasn't been proven
+     * @notice Get proof data for an intent
+     * @param intentHash The intent hash to query
+     * @return ProofData struct containing claimant and destination
      */
-    mapping(bytes32 => address) public provenIntents;
+    function provenIntents(
+        bytes32 intentHash
+    ) external view returns (ProofData memory) {
+        return _provenIntents[intentHash];
+    }
 
     /**
      * @notice Initializes the BaseProver contract
-     * @param _inbox Address of the Inbox contract
+     * @param portal Address of the Portal contract
      */
-    constructor(address _inbox) {
-        INBOX = _inbox;
+    constructor(address portal) {
+        PORTAL = portal;
     }
 
     /**
      * @notice Process intent proofs from a cross-chain message
-     * @param _hashes Array of intent hashes
-     * @param _claimants Array of claimant addresses
+     * @param hashes Array of intent hashes
+     * @param claimants Array of claimant addresses
+     * @param destination Chain ID where the intent is being proven
      */
     function _processIntentProofs(
-        bytes32[] memory _hashes,
-        address[] memory _claimants
+        bytes32[] memory hashes,
+        bytes32[] memory claimants,
+        uint256 destination
     ) internal {
         // If arrays are empty, just return early
-        if (_hashes.length == 0) return;
+        if (hashes.length == 0) return;
 
         // Require matching array lengths for security
-        if (_hashes.length != _claimants.length) {
+        if (hashes.length != claimants.length) {
             revert ArrayLengthMismatch();
         }
 
-        for (uint256 i = 0; i < _hashes.length; i++) {
-            bytes32 intentHash = _hashes[i];
-            address claimant = _claimants[i];
+        for (uint256 i = 0; i < hashes.length; i++) {
+            bytes32 intentHash = hashes[i];
+
+            // Check if the claimant bytes32 represents a valid Ethereum address
+            if (!claimants[i].isValidAddress()) {
+                // Skip non-EVM addresses that can't be converted
+                continue;
+            }
+
+            address claimant = claimants[i].toAddress();
 
             // Validate claimant is not zero address
             if (claimant == address(0)) {
                 continue; // Skip invalid claimants
             }
 
             // Skip rather than revert for already proven intents
-            if (provenIntents[intentHash] != address(0)) {
+            if (_provenIntents[intentHash].claimant != address(0)) {
                 emit IntentAlreadyProven(intentHash);
             } else {
-                provenIntents[intentHash] = claimant;
+                _provenIntents[intentHash] = ProofData({
+                    claimant: claimant,
+                    destination: uint64(destination)
+                });
                 emit IntentProven(intentHash, claimant);
```

Comment:
I thought this event should also be emitted by the Portal according to the specs on prove but I dont see it?